spec/defines/versionlock_spec.rb

```diff
@@ -1,82 +1,184 @@
 require 'spec_helper'
 
 describe 'yum::versionlock' do
-  let(:facts) { { os: { release: { major: 7 } } } }
+  context 'with package_provider set to yum' do
+    let(:facts) do
+      { os: { release: { major: 7 } },
+        package_provider: 'yum' }
+    end
+
+    context 'with a simple, well-formed title 0:bash-4.1.2-9.el6_2.x86_64' do
+      let(:title) { '0:bash-4.1.2-9.el6_2.x86_64' }
+
+      context 'and no parameters' do
+        it { is_expected.to compile.with_all_deps }
+        it { is_expected.to contain_concat__fragment('versionlock_header').with_content("# File managed by puppet\n") }
+        it 'contains a well-formed Concat::Fragment' do
+          is_expected.to contain_concat__fragment("yum-versionlock-#{title}").with_content("#{title}\n")
+        end
+        it { is_expected.to contain_concat('/etc/yum/pluginconf.d/versionlock.list').without_notify }
+      end
+      context 'clean set to true on module' do
+        let :pre_condition do
+          'class { "yum::plugin::versionlock": clean => true, }'
+        end
+
+        it { is_expected.to contain_concat('/etc/yum/pluginconf.d/versionlock.list').with_notify('Exec[yum_clean_all]') }
+        it { is_expected.to contain_exec('yum_clean_all').with_command('/usr/bin/yum clean all') }
+      end
+
+      context 'and ensure set to present' do
+        let(:params) { { ensure: 'present' } }
+
+        it { is_expected.to compile.with_all_deps }
+        it { is_expected.to contain_concat__fragment('versionlock_header').with_content("# File managed by puppet\n") }
+        it 'contains a well-formed Concat::Fragment' do
+          is_expected.to contain_concat__fragment("yum-versionlock-#{title}").with_content("#{title}\n")
+        end
+      end
+
+      context 'and ensure set to absent' do
+        let(:params) { { ensure: 'absent' } }
 
-  context 'with a simple, well-formed title' do
-    let(:title) { '0:bash-4.1.2-9.el6_2.x86_64' }
+        it { is_expected.to compile.with_all_deps }
+        it { is_expected.to contain_concat__fragment('versionlock_header').with_content("# File managed by puppet\n") }
+        it 'contains a well-formed Concat::Fragment' do
+          is_expected.not_to contain_concat__fragment("yum-versionlock-#{title}")
+        end
+      end
+
+      context 'with version set namevar must be just a package name' do
+        let(:params) { { version: '4.3' } }
+
+        it { is_expected.to raise_error(Puppet::PreformattedError, %r{Package name must be formatted as Yum::RpmName, not 'String'}) }
```

Comment:
I've started to like this:
```suggestion
        it { is_expected.to compile.and_raise_error(%r{Package name must be formatted as Yum::RpmName, not 'String'}) }
```

There's a few instances and I won't repeat it everywhere.

Reply:
@ekohl I completly missed these comments before I pressed merge. 

My apologies to excited to merge :-)

I will of course work your comments into a fresh merge request.